src/hooks/use-aspect-ratio-fitting.js

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import useResizeObserver from 'use-resize-observer';
+
+// Returns an offset and scaling factor which, when applied to `element`, will
+// make it properly fit into `container` at the given aspect ratio.
+export default function useAspectRatioFitting({
+  targetWidth = 1366,
+  targetHeight = 768
+}) {
+  const containerRef = React.useRef();
+  const [scaleFactor, setScaleFactor] = React.useState(1);
+  const [transformOrigin, setTransformOrigin] = React.useState({ x: 0, y: 0 });
+
+  const recalculate = React.useCallback(
+    ({ width: containerWidth, height: containerHeight }) => {
+      const containerRatio = containerWidth / containerHeight;
+      const targetRatio = targetWidth / targetHeight;
+      const useVertical = containerRatio > targetRatio;
+
+      const scaleFactor = useVertical
+        ? containerHeight / targetHeight
+        : containerWidth / targetWidth;
+
+      const scaledWidth = targetWidth * scaleFactor;
+      const scaledHeight = targetHeight * scaleFactor;
+
+      let x0 = 0;
+      if (useVertical) {
+        x0 = 0.5 * (containerWidth - scaledWidth);
+        x0 /= 1 - scaleFactor;
+      }
```

Comment:
I was thinking if we wanted to use const we could use a ternary operator here, but I think the way it currently is helps readability